### PR TITLE
Hood upload csv

### PR DESCRIPTION
**Pull request by hectron**
This pull request:

- updates the way that we ingest a neighborhood along with it's house
- creates a rake task to generate a template csv for ingestion
- converts the rake task to accept a csv file relative to the application

### CODE REVIEW

File: lib/tasks/scheduler.rake [hunk 1]
```diff
@@ -40,29 +40,65 @@ task :collect_zpids => :environment do
...
+        city: city,
+        state: state,
+        zip_code: zip_code
+      })
 
-    address = {street_address: street_address, city: city, state: state, zip_code: zip_code}
-    h = House.create(address: address, hood: hood)
-    HouseMetadatum.create(bedrooms: bedrooms, bathrooms: bathrooms, square_feet: square_feet, house: h)
+      HouseMetadatum.create(bedrooms: bedrooms, bathrooms: bathrooms, square_feet: square_feet, house: house)
+    end
```
Comment (hebron-george) [inline lib/tasks/scheduler.rake:101]: Can we kick off the `collect_zpids` task at the end of this one so it fetches all the zpids once the houses have been uploaded?

Or you think we should keep them separate?

Comment (hebron-george) [reply lib/tasks/scheduler.rake:101]: Maybe should be a separate thing - it looks like this task is getting big

Comment (hectron) [reply lib/tasks/scheduler.rake:101]: Separate for now, IMO.

File: lib/tasks/scheduler.rake [hunk 2]
```diff
@@ -40,29 +40,65 @@ task :collect_zpids => :environment do
...
+task :upload_neighborhood, [:hood_onboarding_csv_path] => :environment do |t, args|
   starting_count = House.count
 
-  Rails.logger.info <<~MSG.strip
-    There are #{starting_count} houses in the database.
-    You have passed in #{args.count} arguments, with these keys: #{args.keys}
-  MSG
+  csv_entries = CSV.read(args[:hood_onboarding_csv_path], headers: true).map(&:to_h)
+
+  if csv_entries.flat_map(&:keys).uniq != NeighborhoodCsvHeaders
```
Comment (hebron-george) [inline lib/tasks/scheduler.rake:71]: Is `NeighborhoodCsvHeaders` available in this context? Isn't the constant defined in another task?

Comment (hectron) [reply lib/tasks/scheduler.rake:71]: Yah, the constant is defined at the file level (not in a task).

**Review by hebron-george — APPROVED** [review]

File: lib/tasks/scheduler.rake [hunk 3]
```diff
@@ -40,29 +40,65 @@ task :collect_zpids => :environment do
...
+
+  filename = "#{DateTime.now.to_i}_onboard_neighborhood.csv"
+  CSV.open(filename, "w") { |csv| csv << NeighborhoodCsvHeaders }
+
+  Rails.logger.info "Created #{filename}"
+end
+
 desc 'Upload a CSV of a neighborhood'
-task :upload_neighborhood, [:neighborhood_name, :neighborhood_zip_code, :neighborhood_data] => :environment do |t, args|
+task :upload_neighborhood, [:hood_onboarding_csv_path] => :environment do |t, args|
```
Comment (hectron) [inline lib/tasks/scheduler.rake:66]: This stuff should be moved into a separate rake task, IMO. Can do in a follow up

Comment (hebron-george) [reply lib/tasks/scheduler.rake:66]: What stuff? These lines are two separate tasks?

**Pull request merged**